deepseg.cc

```diff
@@ -190,6 +204,33 @@ void *grab_thread(void *arg) {
 	return NULL;
 }
 
+ModelType get_modeltype(const char* modelname) {
+	if (strstr(modelname, "body-pix")) {
+		return ModelType::BodyPix;
+	}
+	else if (strstr(modelname, "deeplab")) {
+		return ModelType::DeepLab;
+	}
+	else if (strstr(modelname, "segm_")) {
+		return ModelType::GoogleMeetSegmentation;
+	}
+	else if (strstr(modelname, "selfie")) {
+		return ModelType::MLKitSelfie;
+	}
+	fprintf(stderr, "Unknown model type '%s'.\n", modelname);
+	exit(1);
+}
+
+meta_t get_meta(ModelType type) {
+	// TODO: This should be read out from actual mode metadata instead
+	if (type == ModelType::DeepLab) {
+		return meta_t{.std = 127.5, .mean = 127.5};
```

Comment:
Wouldn't be the better description `scaling` and `offset` for those two fields?

`mean` and `std` (st(d)dev) usually have a completely different meaning. Or if those names come from the model files, this should be documented as part of the type declaration for `meta_t`.

Reply:
Yeah I was trying to keep them as in the metadata, but would make more sense with your suggestions. Will change!

Reply:
Actually as we use them (scale, offset) with OpenCV they are different (although derived) from the raw metadata, so different names make sense to me.